packages/pwa-buildpack/lib/WebpackTools/plugins/ServiceWorkerPlugin.js

```diff
@@ -1,34 +1,54 @@
-// TODO: (p1) write test file and test
+const path = require('path');
+const webpack = require('webpack');
 const WorkboxPlugin = require('workbox-webpack-plugin');
 const WriteFileWebpackPlugin = require('write-file-webpack-plugin');
+
+const createFileHash = require('../../Utilities/createFileHash');
 const optionsValidator = require('../../util/options-validator');
 
 // No longer modifiable since v4
-const SW_FILENAME = 'sw.js';
+let SERVICE_WORKER_FILENAME = 'sw.js';
+let SERVICE_WORKER_HASH = '';
+
+const DEFAULT_SW_FILE_PATH = path.join(
+    path.resolve(__dirname).split('/packages')[0],
+    'packages',
+    'venia-concept',
+    'src',
+    'sw.js'
+);
+
+createFileHash(DEFAULT_SW_FILE_PATH).then(hash => {
+    SERVICE_WORKER_HASH = hash;
+    SERVICE_WORKER_FILENAME = `sw.${hash}.js`;
+});
 
 class ServiceWorkerPlugin {
     constructor(config) {
         ServiceWorkerPlugin.validateOptions('ServiceWorkerPlugin', config);
         this.config = config;
     }
+
```

Comment:
Thank you! No sarcasm, this makes it so much more readable 😝

Reply:
It is the prettier extension not me 😲